src/uct/ib/base/ib_md.c

```diff
@@ -190,20 +190,22 @@ extern uct_tl_t UCT_TL_NAME(ud_mlx5);
 extern uct_tl_t UCT_TL_NAME(gga_mlx5);
 
 static uct_tl_t *uct_ib_tls[] = {
-#ifdef HAVE_TL_DC
+#if HAVE_TL_DC
     &UCT_TL_NAME(dc_mlx5),
 #endif
-#ifdef HAVE_TL_RC
+#if HAVE_TL_RC
     &UCT_TL_NAME(rc_verbs),
 #endif
-#if defined (HAVE_TL_RC) && defined (HAVE_MLX5_DV)
+#if HAVE_TL_RC && HAVE_MLX5_DV
     &UCT_TL_NAME(rc_mlx5),
+#if HAVE_DEVX
     &UCT_TL_NAME(gga_mlx5),
 #endif
-#ifdef HAVE_TL_UD
+#endif
+#if HAVE_TL_UD
     &UCT_TL_NAME(ud_verbs),
 #endif
-#if defined (HAVE_TL_UD) && defined (HAVE_MLX5_HW_UD)
+#if HAVE_TL_UD && HAVE_MLX5_HW_UD
```

Comment:
why needed to change?

Reply:
AFAIR `HAVE_XXX` definitions are always defined to `0` or `1` and check it with `#if defined HAVE_XXX` is not correct

Reply:
this seems like unrelated fix that needs another PR?

Reply:
The change regarding to `&UCT_TL_NAME(gga_mlx5)` at line 201 is related and since the patch is small I decided to unite, but OK, I'll split

Reply:
just checked these definitions - they are undefined if built on system without IB or for example, with `--with-devx=no` configure parameter, otherwise they are defined as `1`
Also, C99 standard says that "all remaining identifiers ...  are replaced with the pp-number 0", so it seems to be safe to use `#if HAVE_XXX` as well as `#ifdef`